http/query.go

```diff
@@ -97,10 +103,14 @@ func (r QueryRequest) Validate() error {
 		}
 	}
 
-	if r.Type != "flux" {
+	if r.Type != "flux" && r.Type != "influxql" {
 		return fmt.Errorf(`unknown query type: %s`, r.Type)
 	}
 
+	if r.Type == "influxql" && r.Bucket == "" {
+		return fmt.Errorf("bucket parameter is required for influxql queries")
```

Comment:
with this requiring bucket, could the dbrp mapping be removed?

Reply:
I'm not sure about this API. My understanding was that the API was for connecting the organization and that the bucket would be inferred from the database and retention policy. Is there a way we can avoid adding influxql specific fields to this query request?

Reply:
The bucket is only required for the `/v2/query` endpoint, but its not required for the legacy `/query` endpoint so we can't remove dbrp mappings. Maybe if we remove that legacy endpoint we could, but that would be a separate change.